compiler/compile/OMRSymbolReferenceTable.cpp

```diff
@@ -1899,3 +1900,57 @@ TR::SymbolReference * & OMR::SymbolReferenceTable::element(CommonNonhelperSymbol
       return nullSymRef;
       }
    }
+
+void OMR::SymbolReferenceTable::makeSharedAliases(TR::SymbolReference *sr1, TR::SymbolReference *sr2)
+   {
+   int32_t symRefNum1 = sr1->getReferenceNumber();
+   int32_t symRefNum2 = sr2->getReferenceNumber();
+   TR_BitVector *aliases1 = NULL;
+   TR_BitVector *aliases2 = NULL;
+
+   if (_sharedAliasMap == NULL)
+      {
+      _sharedAliasMap = new (comp()->trHeapMemory()) AliasMap(std::less<int32_t>(), getTypedAllocator<std::pair<int32_t, TR_BitVector *> >(comp()->allocator()));
+      }
+    else
+       {
+       AliasMap::iterator iter1 = _sharedAliasMap->find(symRefNum1);
+       if (iter1 != _sharedAliasMap->end())
+          aliases1 = iter1->second;
+       AliasMap::iterator iter2 = _sharedAliasMap->find(symRefNum2);
+       if (iter2 != _sharedAliasMap->end())
+          aliases2 = iter2->second;
+       }
+
+    if (aliases1 == NULL)
+       {
+       aliases1 = new (comp()->trHeapMemory()) TR_BitVector(self()->getNumSymRefs(), comp()->trMemory(), heapAlloc);
+       aliases1->empty();
+       _sharedAliasMap->insert(std::make_pair(symRefNum1, aliases1));
```

Comment:
I'm not sure how well TR_BitVector copy constructs, but if these fields are canonically part of the map, would it make sense for them physically reside there? In other words, have the map contain actual TR_BitVector objects instead of pointers?